packages/gatsby/src/utils/get-page-data.ts

```diff
@@ -0,0 +1,61 @@
+import * as path from "path"
+import { store, emitter } from "../redux"
+import { IClearPendingPageDataWriteAction } from "../redux/types"
+import * as actions from "../redux/actions/internal"
+import {
+  IPageDataWithQueryResult,
+  readPageData as readPageDataUtil,
+} from "./page-data"
+
+export interface IPageQueryResult {
+  id: string
+  result: IPageDataWithQueryResult
+}
+
+export async function getPageData(pagePath: string): Promise<IPageQueryResult> {
+  const { queries } = store.getState()
+
+  const query = queries.trackedQueries.get(pagePath)
+
+  if (!query) {
+    throw new Error(`Could not find query ${pagePath}`)
+  }
+  if (query.running !== 0) {
+    return waitNextPageData(pagePath)
+  }
+  if (query.dirty !== 0) {
+    store.dispatch(actions.queryRunRequested({ path: pagePath, isPage: true }))
+    return waitNextPageData(pagePath)
+  }
```

Comment:
I think this should be commented out for now and just uncommented when we have code to trigger state transitions, otherwise we can end up with never resolved code when using `GATSBY_EXPERIMENTAL_GET_PAGE_DATA`?

Reply:
Actually - I think we should just include the transition in this PR, so this is complete, I will work on this